backend/src/main/java/com/bakdata/conquery/models/forms/util/DateContext.java

```diff
@@ -121,53 +116,50 @@
 	 * @param timeUnit
 	 * @return
 	 */
-	public static List<DateContext> generateRelativeContexts(int event, IndexPlacement indexPlacement, int featureTime,
-			int outcomeTime, boolean sliced, TimeUnit timeUnit) {
+	public static List<DateContext> generateRelativeContexts(int event, IndexPlacement indexPlacement, int featureTime,	int outcomeTime, DateContextMode timeUnit, List<DateContextMode> subdivisionModes) {
 		if (featureTime < 1 || outcomeTime < 1) {
 			throw new IllegalArgumentException("Relative times were smaller than 1 (featureTime: " + featureTime
 					+ "; outcomeTime: " + outcomeTime + ")");
 		}
-		List<DateContext> dcl = new ArrayList<>();
+		List<DateContext> dcList = new ArrayList<>();
+		
+		LocalDate eventdate = CDate.toLocalDate(event);
 
 		CDateRange featureRange = generateFeatureRange(event, indexPlacement, featureTime, timeUnit);
 		CDateRange outcomeRange = generateOutcomeRange(event, indexPlacement, outcomeTime, timeUnit);
 
-		dcl.add(new DateContext(featureRange, FeatureGroup.FEATURE, null, CDate.toLocalDate(event)));
-		dcl.add(new DateContext(outcomeRange, FeatureGroup.OUTCOME, null, CDate.toLocalDate(event)));
-
-		if (sliced) {
-			List<CDateRange> featureRanges = null;
-			List<CDateRange> outcomeRanges = null;
 
-			switch(timeUnit) {
-				case DAYS:
-					featureRanges = featureRange.getCoveredDays();
-					outcomeRanges = outcomeRange.getCoveredDays();
-					break;
-				case QUARTERS:
-					featureRanges = featureRange.getCoveredQuarters();
-					outcomeRanges = outcomeRange.getCoveredQuarters();
-					break;
-				default:
-					throw new IllegalArgumentException("Resolution " + timeUnit + " not supported.");				
-			}
-
-			int numRanges = featureRanges.size();
-			int idx = indexPlacement.equals(IndexPlacement.BEFORE) ? numRanges - 1 : numRanges;
-			for (CDateRange range : featureRanges) {
-				dcl.add(new DateContext(range, FeatureGroup.FEATURE, -idx, CDate.toLocalDate(event)));
-				idx--;
+		for(DateContextMode mode : subdivisionModes) {
+			List<CDateRange> featureRanges = mode.subdivideRange(featureRange);
+			int index = indexPlacement.equals(IndexPlacement.BEFORE) ? featureRanges.size() - 1 : featureRanges.size();
+			for (CDateRange subRange : featureRanges) {
+				DateContext dc = new DateContext(
+					subRange,
+					FeatureGroup.FEATURE,
+					// For now there is no index for complete
+					mode.equals(DateContextMode.COMPLETE)? null : -index,
```

Comment:
Uh, ternary works even without a space in front of the question mark?

Reply:
Ach, die diskussion um whitespaces hab ich schon lange aufgegeben :D Wenn sie schön sind sind sie schön, wenn nicht werden sie beim nächsten anfassen schön :D

🗡 🗡 🗡 🗡

Reply:
In the Javascript world we don't even discuss this, we just use the code formatter https://prettier.io/ to auto-format on save. Is there something like this in the Java land?

Reply:
Wir hatten ein paar Probleme damit, weil die Formatting Plugins der verschiedenen IDEs sich nicht exakt gleich verhalten und die Diffs immer sehr groß waren.

Reply:
Meine Einstellung, dass nur die Zeilen formiert werden, die ich angefasst habe ist verloren gegangen